dbt/adapters/databricks/impl.py

```diff
@@ -61,6 +61,12 @@ class DatabricksConfig(AdapterConfig):
     tblproperties: Optional[Dict[str, str]] = None
 
 
+def check_not_found_error(errmsg: str) -> bool:
+    new_error = "[SCHEMA_NOT_FOUND]" in errmsg
+    old_error = re.match(r".*(Database).*(not found).*", errmsg, re.DOTALL)
+    return new_error or old_error is not None
```

Comment:
nit: `new_error or (old_error is not None)` to make it clear?

Reply:
I disagree with ueshin's suggestion but agree with the lack of clarity.

The way this code is written right now, if `new_error` is not `None` _it will be returned by the function!_ (which means the type annotation is incorrect because it would return a string instead of a bool).

To guarantee that the function always returns a boolean the code should be:

```python
return (new_error or old_error) is not None
```

This is because in Python, a truthy predicate to `or` _will be returned!_. So you can write fancy things like:

```python
return new_error and new_error
````

Which would return the actual error if it is present or `False` if it isn't.

The issue with ueshin's suggestion is that `new_error or (old_error is not None)` will still return the string text of `new_error` instead of a bool in cases where `new_error` is not Null.

Reply:
`new_error` is boolean so it won't return the string test.

We could also do:
```py
new_error = "[SCHEMA_NOT_FOUND]" in errmsg
old_error = re.match(r".*(Database).*(not found).*", errmsg, re.DOTALL) is not None
return new_error or old_error
```

Reply:
Well, if this is the kind of discussion this PR is generating, it seems the fix is okay

Reply:
My mistake! `new_error` is absolutely a bool with the code as written. I was reading it too fast. I like @ueshin 's final proposal as it is the most clear.